plugins/tuya/src/camera.ts

```diff
@@ -0,0 +1,209 @@
+        this.setStatusIndicator(true);
+    }
+
+    private async setStatusIndicator(on: boolean) {
+        const camera = this.findCamera();
+        const statusIndicator = TuyaDevice.getStatusIndicator(camera);
+
+        if (statusIndicator) {
+            await this.controller.api.updateDevice(camera, [
+                {
+                    code: statusIndicator.code,
+                    value: on
+                }
+            ]);
+        }
+    }
+
+    // VideoCamera
+
+    async getVideoStream(
+        options?: MediaStreamOptions
+    ): Promise<MediaObject> {
+        const vsos = await this.getVideoStreamOptions();
+        const vso = vsos.find(find => find.id === options.id) || vsos[0];
+
+        // Always create new rtsp since it can only be used once and we only have 30 seconds before we can
+        // use it.
+
+        const camera = this.findCamera();
+
+        if (!camera) {
+            this.logger.w(`Could not find camera for ${this.name} to show stream.`);
+            throw new Error(`Failed to stream ${this.name}: Camera not found.`);
+        }
+
+        if (!camera.online) {
+            this.logger.w(`${this.name} is currently offline. Will not be able to stream until device is back online.`);
+            throw new Error(`Failed to stream ${this.name}: Camera is offline.`);
+        }
+
+        const rtsps = await this.controller.api.getRTSPS(camera);
+
+        if (!rtsps) {
+            this.logger.w("There was an error retreiving camera's rtsps for streamimg.");
+            throw new Error(`Failed to capture stream for ${this.name}: RTSPS link not found.`);
+        }
+
+        const mediaStreamUrl: MediaStreamUrl = {
+            url: rtsps.url,
+            container: 'rtsp',
+            mediaStreamOptions: vso
+        }
+        return this.createMediaObject(mediaStreamUrl, ScryptedMimeTypes.MediaStreamUrl);
+    }
+
+    async getVideoStreamOptions(): Promise<ResponseMediaStreamOptions[]> {
+        return [
+            {
+                id: 'default',
+                container: 'rtsp',
+                video: {
+                    codec: 'h264',
+                },
+                audio: {
+                    codec: 'pcm_ulaw'
+                },
+                source: 'cloud',
+                tool: 'scrypted',
+                userConfigurable: false
+            }
+        ];
+    }
+
+    // Motion
+
+    // most cameras have have motion and doorbell press events, but dont notify when the event ends.
+    // so set a timeout ourselves to reset the state.
+
+    triggerBinaryState() {
+        this.binaryState = true;
+        setTimeout(() => this.binaryState = false, 10000);
+    }
+
+    // most cameras have have motion and doorbell press events, but dont notify when the event ends.
+    // so set a timeout ourselves to reset the state.
+
+    triggerMotion() {
+        this.motionDetected = true;
+        setTimeout(() => this.motionDetected = false, 10000);
```

Comment:
this will have funny behavior if motion is triggered multiple times, should maybe use a debounce of some sort to keep the motion active.

Reply:
I think this was a bug I had in my camera plugins if that's where you got the code, but it was recently fixed in most of them

Reply:
Oops, I actually had that in mind but forgot to add it